### PR TITLE
VSCE: Add help and feedback sidebar 

### PR DESCRIPTION
**Pull request by abeatrix**
Continue from https://github.com/sourcegraph/sourcegraph/pull/32586

Closes: https://github.com/sourcegraph/sourcegraph/issues/31021, https://github.com/sourcegraph/sourcegraph/issues/32741, https://github.com/sourcegraph/sourcegraph/issues/26536

This PR adds a new sidebar view called Help  and Feedback that is collapsed by default:
Screen Shot 2022-03-17 at 2 32 11 PM

Screen Shot 2022-03-17 at 2 59 26 PM

We have also moved the devdependencies from the VSCE directory to the root file for better developer experience and added a CONTRIBUTING.md file.

## Test plan

Added `yarn build-vsce` and `yarn watch-vsce`

### CODE REVIEW

File: client/vscode/src/webview/sidebars/help/HelpSidebarView.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}) => {
+    const [hasAccount, setHasAccount] = useState(false)
+
+    const onHelpItemClick = async (url: string): Promise<void> => {
+        platformContext.telemetryService.log('VSCESidebarCreateAccount')
+        await extensionCoreAPI.openLink(url)
+    }
+
+    return (
+        // const [state, setState] = useState<'initial' | 'validating' | 'success' | 'failure'>('initial')
```
Comment (tjkandala) [inline client/vscode/src/webview/sidebars/help/HelpSidebarView.tsx:30]:
```suggestion
```

File: client/vscode/src/webview/sidebars/help/HelpSidebarView.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+export const HelpSidebarView: React.FunctionComponent<HelpSidebarViewProps> = ({
+    platformContext,
+    extensionCoreAPI,
+    authenticatedUser,
+    instanceURL,
+}) => {
+    const [hasAccount, setHasAccount] = useState(false)
+
+    const onHelpItemClick = async (url: string): Promise<void> => {
+        platformContext.telemetryService.log('VSCESidebarCreateAccount')
```
Comment (tjkandala) [inline client/vscode/src/webview/sidebars/help/HelpSidebarView.tsx:25]: Shouldn't we not log this same event for all the different types of links (example: for feedback link)?

File: client/vscode/src/webview/sidebars/auth/AuthSidebarView.tsx [hunk 3]
```diff
@@ -8,16 +8,18 @@ import { currentAuthStateQuery } from '@sourcegraph/shared/src/auth'
...
-import { WebviewPageProps } from '../platform/context'
+import { WebviewPageProps } from '../../platform/context'
 
 import styles from './AuthSidebarView.module.scss'
 
 const SIDEBAR_UTM_PARAMS = 'utm_medium=VSCIDE&utm_source=sidebar&utm_campaign=vsce-sign-up&utm_content=sign-up'
 
-interface AuthSidebarViewProps extends WebviewPageProps {
+interface AuthSidebarViewProps extends Pick<WebviewPageProps, 'extensionCoreAPI' | 'platformContext' | 'instanceURL'> {
     stateStatus: string
```
Comment (tjkandala) [inline client/vscode/src/webview/sidebars/auth/AuthSidebarView.tsx:18]:
```suggestion
    stateStatus: VSCEState['status']
```

We can reference the string union type directly (gets us more type safety, better autocomplete)

File: client/vscode/src/webview/sidebars/auth/AuthSidebarView.tsx [hunk 4]
```diff
@@ -28,7 +30,7 @@ export const AuthSidebarView: React.FunctionComponent<AuthSidebarViewProps> = ({
     stateStatus,
 }) => {
     const [state, setState] = useState<'initial' | 'validating' | 'success' | 'failure'>('initial')
-    const [hasAccount, setHasAccount] = useState(false)
+    const [hasAccount, setHasAccount] = useState(!stateStatus)
```
Comment (tjkandala) [inline client/vscode/src/webview/sidebars/auth/AuthSidebarView.tsx:33]: Isn't it possible that `stateStatus` can be defined while a user has an account?

**Review by tjkandala — APPROVED** [review]
Awesome refactoring!

File: client/vscode/src/webview/sidebars/help/HelpSidebarView.tsx (hunk 2 above)
Comment (abeatrix) [reply client/vscode/src/webview/sidebars/help/HelpSidebarView.tsx:25]: Yes, thanks for catching that! Meant to do it but totally forgot about it 😆  It is updated in the latest commit (not pushed yet)!

File: client/vscode/src/webview/sidebars/auth/AuthSidebarView.tsx (hunk 4 above)
Comment (abeatrix) [reply client/vscode/src/webview/sidebars/auth/AuthSidebarView.tsx:33]: True! Updated to use authenticatedUser from WebviewProps instead !

**Pull request merged**